### PR TITLE
Ticket 17

### PR DESCRIPTION
**Pull request by kipp0**
map receives address from frontend and spits out coordinates and bounds

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[[source]]
```
Comment (sina-jamshidi) [inline Pipfile]: It looks like these were made in the root directory. You should ideally run your virtualenv from the server directory to make sure you're adding things to the right pipfile and running things properly. So you can delete this `Pipfile` and `Pipfile.lock`

File: server/api/map_handler.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import smtplib
+import requests
+from flask import jsonify, request, Blueprint
+from config import API_KEY
+map_handler = Blueprint('map_handler', __name__)
+
+@map_handler.route('/map', methods=['POST'])
+def map():
+    if request.method != 'POST':
+        return 'NOT POST'
```
Comment (sina-jamshidi) [inline server/api/map_handler.py]: I think Flask already handles this validation for you

File: server/api/map_handler.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    
+    postData = json.loads(request.get_data())
```
Comment (sina-jamshidi) [inline server/api/map_handler.py]: In Python, function and variable names are in `snake_case`. Class names are in `PascalCase`. [See details](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names)

File: server/api/map_handler.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    address = postData['address'].replace(' ', '+')
+    
+    url = "https://maps.googleapis.com/maps/api/geocode/json?address=%s&key=%s" % (address, API_KEY)
+    googleReq = requests.get(url).json()
+
+    #check for bad request
+    #check if location exists
+    googleHandler = googleAPIHandler()
+    status = googleHandler.handle(googleReq)
+    if status != True:
```
Comment (sina-jamshidi) [inline server/api/map_handler.py]:
```suggestion
    if not status:
```

File: server/api/map_handler.py (hunk 3 above)
Comment (sina-jamshidi) [inline server/api/map_handler.py]: I think you may be able to use `to_dict()` instead of `json.loads`

File: server/api/map_handler.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        print(status['log'])
+        return status['msg']
+    
+    #get geocodes
+    geocodes = googleReq['results'][0]['geometry']['location']
+    #get bounds: viewport -> latitude,longitude values defining the southwest and northeast corner of the viewport bounding box
+    viewport = googleReq['results'][0]['geometry']['viewport']
+    
+    #return geolocation
+    return {'geocodes':geocodes,'viewport':viewport}
```
Comment (sina-jamshidi) [inline server/api/map_handler.py]: @SilverStrings024 is using `make_response` to generate the responses. You guys should discuss and decide on one way to handle returning responses so the code is more uniform. Does this even work properly though? Generally you have to do some kind of `jsonify(response)` or something.

File: server/api/map_handler.py (hunk 4 above)
Comment (sina-jamshidi) [reply server/api/map_handler.py]: Or `if status is not True:`.
Both are the more "pythonic" way of doing this check

File: server/api/map_handler.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+class googleAPIHandler:
```
Comment (sina-jamshidi) [inline server/api/map_handler.py]: `GoogleAPIHandler`

File: server/api/map_handler.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    
+    gMapRequest = {}
```
Comment (sina-jamshidi) [inline server/api/map_handler.py]: `google_maps_request`.
You probably want to put this in some kind of `init` function and reference it as `self.google_maps_request`

**Review by sina-jamshidi — CHANGES_REQUESTED** [review]
Looks close to ready

File: server/api/map_handler.py (hunk 4 above)
Comment (kipp0) [reply server/api/map_handler.py]: will if not status be true if it is not undefined or null variable

Comment (kipp0) [reply server/api/map_handler.py]: I'll just change it to if status != 'OK' in order to prevent confusion

File: server/api/map_handler.py (hunk 5 above)
Comment (kipp0) [reply server/api/map_handler.py]: I've changed it

File: server/api/map_handler.py (hunk 7 above)
Comment (kipp0) [reply server/api/map_handler.py]: ok I thought about doing that. I'll do it now

File: Pipfile (hunk 1 above)
Comment (kipp0) [reply Pipfile]: Ticket 17 - PR changes is up and ready for review

File: server/api/map_handler.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    url = "https://maps.googleapis.com/maps/api/geocode/json?address=%s&key=%s" % (address, API_KEY)
+    google_req = requests.get(url).json()
+
+    #check for bad request
+    #check if location exists
+    googleHandler = GoogleAPIHandler(google_req)
+    status = googleHandler.handle()
+    if status != 'OK':
+        print(status['log'])
+        return status['msg'], 400
```
Comment (dproc96) [inline server/api/map_handler.py:23]:
```suggestion
        return jsonify({"message": status['msg']}), 400
```

**Review by dproc96 — APPROVED** [review]
Had one small change request then this is good to go

**Pull request merged**